packages/flutter/lib/src/painting/box_border.dart

```diff
@@ -463,7 +463,7 @@ class Border extends BoxBorder {
 
   @override
   EdgeInsetsGeometry get dimensions {
-    if (_widthIsUniform) {
+    if (_widthIsUniform && _strokeAlignIsUniform) {
```

Comment:
Why do we need to distinguish between the uniform and non-uniform cases? Can we just all use the non-uniform constructors?

Reply:
Theoretically it is an optimization so you save 3 getters, each has a 4 step math operation, so you are saving 12 calculations per render. Probably affect nothing, but this is run every time everywhere, so I tried to optimize.

If the code is ugly, we could un-optimize, but I don't know, feels fine either way..

Reply:
Rather than whether the code is pretty, I think what's more important is the maintainability. You see, why this bug would happen at all is because we fail to count in all factors that make `left.strokeInset` equal to `top.strokeInset`, which allows us to use `top.strokeInset` for all four. But what if more factors are added to `strokeInset` in the future? We'll need to remember to update this as well. In general, it feels like this "shortcut" relies on implementation details.

In terms of efficiency, I don't think you're saving much either. You'll have to read all factors to fully determine the equality anyway, therefore you can't save reading any getters.

In conclusion, I don't think the performance saved here is worth the risk of maintainability.

Reply:
Ok, feel free to do this change @romaingyh

Reply:
@bernaferrari done :)